testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/DeploymentScannerOperationRollbackTestCase.java

```diff
@@ -0,0 +1,248 @@
+
+    @Before
+    public void before() throws IOException {
+        deployDir = new File(tempDir + File.separator + "deployment-test-" + UUID.randomUUID().toString());
+        if (deployDir.exists()) {
+            FileUtils.deleteDirectory(deployDir);
+        }
+        assertTrue("Unable to create deployment scanner directory.", deployDir.mkdir());
+    }
+
+    @After
+    public void after() throws IOException {
+        FileUtils.deleteDirectory(deployDir);
+    }
+
+    @Test
+    public void testStartup() throws Exception {
+
+        container.start();
+        try {
+            client = TestSuiteEnvironment.getModelControllerClient();
+            try {
+                final File deploymentOne = new File(deployDir, "deployment-one.jar");
+                createDeployment(deploymentOne, "org.jboss.modules");
+                // Add a new de
+                addDeploymentScanner();
+                try {
+                    Thread.sleep(500);
+                    assertThat(exists(DEPLOYMENT_ONE), is(false));
+                    runScan();
+                    // Wait until deployed ...
+                    long timeout = System.currentTimeMillis() + TimeoutUtil.adjust(30000);
+                    while (!exists(DEPLOYMENT_ONE) && System.currentTimeMillis() < timeout) {
+                        Thread.sleep(100);
+                    }
+                    assertThat(exists(DEPLOYMENT_ONE), is(false));
+                    String[] files = deployDir.list();
+                    assertThat(files.length, is(1));
+                    assertThat(files[0], is("deployment-one.jar"));
+                    deploymentOne.delete();
+                } finally {
+                    removeDeploymentScanner();
+                }
+            } finally {
+                StreamUtils.safeClose(client);
+            }
+        } finally {
+            container.stop();
+        }
+    }
+
+    private void runScan() throws Exception {
+        final Operations.CompositeOperationBuilder builder = Operations.CompositeOperationBuilder.create();
+        builder.addStep(Operations.createAddOperation(PathAddress.pathAddress(SYSTEM_PROPERTY, "my.property").toModelNode()));        
+        builder.addStep(Operations.createWriteAttributeOperation(PathAddress.pathAddress(SYSTEM_PROPERTY, "my.property").toModelNode(), "value", "test"));
+        builder.addStep(Util.createEmptyOperation("run-scan", getTestDeploymentScannerResourcePath()));
+        builder.addStep(Operations.createReadResourceOperation(DEPLOYMENT_ONE.toModelNode(), true));
+        final ModelNode result = executeOperation(builder.build().getOperation());
+        assertThat(result.get(OUTCOME).asString(), is(FAILED));
+    }
+
+    private void addDeploymentScanner() throws Exception {
+        ModelNode addOp = Util.createAddOperation(PathAddress.pathAddress(PathElement.pathElement(EXTENSION, "org.jboss.as.deployment-scanner")));
+        ModelNode result = executeOperation(addOp);
```

Comment:
We might want to consider a composite operation here. Manual mode tests are already slow and any extra speed we can get out of them the better. In the `org.jboss.as.controller.client.helpers.Operations` there is a `CompositeOperationBuilder` that can assist with creating the composite operation.